### PR TITLE
VSTS-281 fix release and promotion

### PR DESCRIPTION
**Pull request by mickael-caro-sonarsource**
This PR's intent is : 

- Split SQ analysis for both SQ and SC extensions
- Fix SQ analysis version number
- Split few gulp task
- Fix SQ/SC extension versions in repox
- Revamp directory structure on repox for extensions

In preparation of future GH action release deployment (VSTS-279)

Please review our [contribution guidelines](https://github.com/SonarSource/sonar-scanner-vsts/blob/master/contributing.md).

And please ensure your pull request adheres to the following guidelines: 

- [ ] Please explain your motives to contribute this change: what problem you are trying to fix, what improvement you are trying to make
- [ ] Use the following formatting style: [SonarSource/sonar-developer-toolset](https://github.com/SonarSource/sonar-developer-toolset#code-style)
- [ ] Provide a unit test for any code you changed
- [ ] If there is a [JIRA](https://jira.sonarsource.com/browse/VSTS) ticket available, please make your commits and pull request start with the ticket ID (VSTS-XXXX)

### CODE REVIEW

File: common/ts/publish-task.ts [hunk 1]
```diff
@@ -79,7 +79,12 @@ export async function getReportForTask(
   } catch (e) {
     if (e instanceof TimeOutReachedError) {
       tl.warning(
-        `Task '${taskReport.ceTaskId}' takes too long to complete. Stopping after ${timeoutSec}s of polling. No quality gate will be displayed on build result.`
+        "Task '" +
```
Comment (gregaubert) [inline common/ts/publish-task.ts:82]: I'm wondering why this change? It's the same as before but harder to read?

File: commonv5/ts/publish-task.ts [hunk 2]
```diff
@@ -79,7 +79,12 @@ export async function getReportForTask(
   } catch (e) {
     if (e instanceof TimeOutReachedError) {
       tl.warning(
-        `Task '${taskReport.ceTaskId}' takes too long to complete. Stopping after ${timeoutSec}s of polling. No quality gate will be displayed on build result.`
+        "Task '" +
```
Comment (gregaubert) [inline commonv5/ts/publish-task.ts:82]: Same

File: config/utils.js [hunk 3]
```diff
@@ -154,10 +178,11 @@ exports.runSonnarQubeScanner = function(callback, options = {}) {
...
       options: {
+        ...customOptions,
         ...commonOptions,
         ...options
       }
     },
     callback
   );
-};
+}
```
Comment (gregaubert) [inline config/utils.js:188]: Missing EOF new line

File: common/ts/publish-task.ts (hunk 1 above)
Comment (mickael-caro-sonarsource) [reply common/ts/publish-task.ts:82]: For some reason few issues have been raised on next regarding line length, that's why I needed to wrap it (I also enforced it in the eslintrc file)

Comment (gregaubert) [reply common/ts/publish-task.ts:82]: I would typically close as won't fix such an error, because in this case, fixing the error makes the code less maintainable than before...

Comment (gregaubert) [reply common/ts/publish-task.ts:82]: Or you could eventually fix it in a cleaner way, maybe like:
```ts
      tl.warning(
        `Task '${taskReport.ceTaskId}' takes too long to complete. `
        `Stopping after ${timeoutSec}s of polling. No quality gate will be displayed on build result.`
```

Comment (mickael-caro-sonarsource) [reply common/ts/publish-task.ts:82]: Ok that makes sense and i do agree thanks !
